File: java/src/DirectionEast.java

public class DirectionEast implements IDirection
{
	public IDirection left()
	{
		return new DirectionNorth();
	}
	
	public IDirection right()
	{
		return new DirectionSouth();
	}
	
	public Position move(Position position)
	{
		position.setXPosition(position.getXPosition() + 1 );
		return position;
	}

	public String name()
	{
		return "EAST";
	}
}

File: ruby/src/DirectionEast.rb
require_relative "IDirection.rb"
require_relative "DirectionNorth.rb"
require_relative "DirectionSouth.rb"

class DirectionEast < IDirection
  
  def name()
    return "EAST"
  end
  
  def left()
    return DirectionNorth.new()
  end
  
  def right()
    return DirectionSouth.new()
  end
  
  def move(position)
    position.x_position = position.x_position + 1
    return position
  end
end
